enrichment-mongo/src/main/java/eu/europeana/enrichment/mongo/service/PersistentTopicEntityServiceImpl.java

```diff
@@ -0,0 +1,48 @@
+package eu.europeana.enrichment.mongo.service;
+
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import eu.europeana.enrichment.common.commons.AppConfigConstants;
+import eu.europeana.enrichment.model.TopicEntity;
+import eu.europeana.enrichment.model.TopicModel;
+import eu.europeana.enrichment.model.impl.TopicEntityImpl;
+import eu.europeana.enrichment.mongo.dao.TopicEntityDao;
+
+@Service(AppConfigConstants.BEAN_ENRICHMENT_PERSISTENT_TOPIC_ENTITY_SERVICE)
+public class PersistentTopicEntityServiceImpl implements PersistentTopicEntityService{
+	
+	
+	@Autowired
+	private TopicEntityDao topicEntityDao;
+
+	@Override
+	public void saveTopicEntity(TopicEntity topicEntity) {
+		topicEntityDao.saveTopicEntity(topicEntity);
+	}
+
+	@Override
+	public TopicEntity findTopicEntityByIdentifier(String topicIdentifier) {
+		return topicEntityDao.findTopicEntityByIdentifier(topicIdentifier);
+	}
+
+	@Override
+	public void updateTopicEntity(TopicEntity dbtopicEntity) {
```

Comment:
The update methods seem not to be needed, the save can be used instead. Just make sure the entity passed to be saved is the one from the db, if it exists there. See some code here: https://github.com/EnrichEuropeana/enrichment/blob/55d8d8bdb5a5c03d5e3d573801689678ea7ae9b7/enrichment-web/src/main/java/eu/europeana/enrichment/web/service/impl/EnrichmentNERServiceImpl.java#L299. If the entity to be saved/updated already exists in the db, we first fetch it from there, modify it, and save again. In that way, the same entity will be updated in the db.

Reply:
there is a difference between save and update. See DAO methods

Reply:
eventualy throw exception if the ObjectId is not present in the TopicEntity (as that is a create request)

Reply:
rename to Update and throw excepttion if mongo ObjectId is null to prevent dupplicates